packages/@lwc/rollup-plugin/src/__tests__/index.spec.js

```diff
@@ -76,6 +77,16 @@ describe('rollup in compat mode', () => {
     });
 });
 
+describe('typescript relative import', () => {
+    it(`should resolve to .ts file`, () => {
+        const entry = path.join(tsAppDir, 'main.ts');
+        return doRollup(entry, { compat: false }).then(({ code: actual }) => {
+            const expected = fsExpected('expected_default_config_ts_simple_app');
+            expect(pretty(actual)).toBe(pretty(expected));
+        });
+    });
+});
+
```

Comment:
i would like to see a mixed case of js importing ts or vice versa to ensure the scenario is handled correctly

Reply:
We should actually throw if you are have a mix.

Reply:
That would definitely simplify things... I wouldn't go as far as saying it makes "no sense" since mixing TS and JS is pretty common in brownfield projects that are trying to move to typescript slowly - after all, the typescript compiler has an option `allowJs` so it's very much a supported scenario from Typescripts point-of-view.

Having said that, mixing TS and JS is more of a feature to consider adding in the future if there is demand... the goal of this PR is to simply fix the bug of "my TS file cannot relatively import another TS file".  So I like the idea for now of just throwing if TS and JS are mixed.  I'll update the PR.

Reply:
If you are using LWC you are writing new things, so there is not upgrade path :) I think the logic is fine as it is but if you can throw when the missmatch  happens it will avoid us to have to support that case all togeher,